plugins/de.cau.cs.kieler.klighd.lsp/src/de/cau/cs/kieler/klighd/lsp/utils/RenderingPreparer.xtend

```diff
@@ -0,0 +1,442 @@
+/*
+ * KIELER - Kiel Integrated Environment for Layout Eclipse RichClient
+ *
+ * http://rtsys.informatik.uni-kiel.de/kieler
+ * 
+ * Copyright 2018 by
+ * + Kiel University
+ *   + Department of Computer Science
+ *     + Real-Time and Embedded Systems Group
+ * 
+ * This code is provided under the terms of the Eclipse Public License (EPL).
+ */
+package de.cau.cs.kieler.klighd.lsp.utils
+
+import com.google.common.base.Predicate
+import com.google.common.base.Strings
+import de.cau.cs.kieler.klighd.IStyleModifier.StyleModificationContext
+import de.cau.cs.kieler.klighd.KlighdDataManager
+import de.cau.cs.kieler.klighd.kgraph.KEdge
+import de.cau.cs.kieler.klighd.kgraph.KGraphElement
+import de.cau.cs.kieler.klighd.kgraph.KLabeledGraphElement
+import de.cau.cs.kieler.klighd.kgraph.KNode
+import de.cau.cs.kieler.klighd.kgraph.KPoint
+import de.cau.cs.kieler.klighd.kgraph.KShapeLayout
+import de.cau.cs.kieler.klighd.krendering.KAreaPlacementData
+import de.cau.cs.kieler.klighd.krendering.KContainerRendering
+import de.cau.cs.kieler.klighd.krendering.KDecoratorPlacementData
+import de.cau.cs.kieler.klighd.krendering.KGridPlacement
+import de.cau.cs.kieler.klighd.krendering.KPlacement
+import de.cau.cs.kieler.klighd.krendering.KPointPlacementData
+import de.cau.cs.kieler.klighd.krendering.KPolygon
+import de.cau.cs.kieler.klighd.krendering.KPolyline
+import de.cau.cs.kieler.klighd.krendering.KRendering
+import de.cau.cs.kieler.klighd.krendering.KRenderingLibrary
+import de.cau.cs.kieler.klighd.krendering.KRenderingRef
+import de.cau.cs.kieler.klighd.krendering.KStyle
+import de.cau.cs.kieler.klighd.microlayout.Bounds
+import de.cau.cs.kieler.klighd.microlayout.DecoratorPlacementUtil
+import de.cau.cs.kieler.klighd.microlayout.DecoratorPlacementUtil.Decoration
+import de.cau.cs.kieler.klighd.microlayout.GridPlacementUtil
+import de.cau.cs.kieler.klighd.microlayout.PlacementUtil
+import java.awt.geom.Point2D
+import java.util.ArrayList
+import java.util.HashMap
+import java.util.List
+import java.util.Map
+
+import static com.google.common.collect.Iterables.filter
+
+/**
+ * Utility class to provide some functionality to persist prepare the rendering of a {@link KGraphElement}.
+ * 
+ * @author nre
+ */
+final class RenderingPreparer {
```

Comment:
This class seems to include functions that may also be part of the "normal" KLighD stuff. Can we reuse that here?

Reply:
Similar to the microlayout of the Piccolo viewer being done in the piccolo package, this Sprotty-specific micro layout size estimation is done here. I guess some parts could be externalized and reused for both viewers though.

Reply:
Should this be part of the next release or is this too complicated?